contrib/corrosion-cmake/CMakeLists.txt

```
@@ -0,0 +1,45 @@
+if (NOT ENABLE_LIBRARIES)
+  option(ENABLE_RUST "Enable rust" FALSE)
+elseif((CMAKE_TOOLCHAIN_FILE MATCHES "darwin") AND (CMAKE_TOOLCHAIN_FILE MATCHES "aarch64"))
+  option(ENABLE_RUST FALSE)
+  message(STATUS "Rust is not available on aarch64-apple-darwin")
+  return()
+else()
+  list (APPEND CMAKE_MODULE_PATH "${ClickHouse_SOURCE_DIR}/contrib/corrosion/cmake")
+  find_package(Rust)
+  option(ENABLE_RUST "Enable rust" ${Rust_FOUND})
+endif()
```

```suggestion
if (NOT ENABLE_LIBRARIES)
  set(DEFAULT_ENABLE_RUST FALSE)
elseif((CMAKE_TOOLCHAIN_FILE MATCHES "darwin") AND (CMAKE_TOOLCHAIN_FILE MATCHES "aarch64"))
  message(STATUS "Rust is not available on aarch64-apple-darwin")
  set(DEFAULT_ENABLE_RUST FALSE)
else()
  list (APPEND CMAKE_MODULE_PATH "${ClickHouse_SOURCE_DIR}/contrib/corrosion/cmake")
  find_package(Rust)
  set(DEFAULT_ENABLE_RUST ${Rust_FOUND})
endif()

option(ENABLE_RUST "Enable rust" ${DEFAULT_ENABLE_RUST})
```

seems a bit better because this way `option(ENABLE RUST ...)` is written only once